docs/design/snapshot.md

```diff
@@ -0,0 +1,227 @@
+Snapshots are read-only copies of volumes (once the snapshot is activated) that
+can be used for cloning new volumes from. Its is possible to create a snapshot
+through the [Snapshot Create API](#create-a-snapshot) or the commandline
+client.
+
+From the command line client, you can type the following to create a snapshot
+from an existing volume:
+
+```
+$ heketi-cli volume snapshot <vol_uuid> [-name=<snap_name>]
+```
+
+The new snapshot can be used to create a new volume:
+
+```
+$ heketi-cli snapshot clone <snap_uuid> [-name=<clone_name>]
+```
+
+The clones of snapshots are new volumes with the same properties as the
+original. The cloned volumes can be deleted with the `heketi-cli volume delete`
+command. In a similar fashion, snapshots can be removed with the `heketi-cli
+snapshot delete` command.
+
+
+## Implementation Detail for Heketi internal Snapshot objects
+The proposed API and CLI can be used for both file-volumes and block-volumes.
+In order to provide the expected outcome of the cloning, the Heketi internal
+snapshot object should have a `Type: {file-volume|block-volume}`. This makes it
+possible to have a single `/snapshots/{snapshot_uuid}` endpoint that can handle
+both volume types.
+
+
+# Proposed CLI
+
+```
+$ heketi-cli volume snapshot <vol_uuid> [--name=<snap_uuid>] [--description=<string>]
+$ heketi-cli snapshot clone <snap_uuid> [--name=<vol_uuid>]
+$ heketi-cli snapshot delete <snap_uuid>
+$ heketi-cli snapshot list
+$ heketi-cli snapshot info <snap_uuid>
+```
+
+Convenience call for direct volume cloning:
+
+```
+$ heketi-cli volume clone <vol_uuid>
+```
+
+
+# API Proposal
+
+The API is layed out here for file volume types only. The same API will be
+added for block-volumes at a later time.
+
+### Create a Snapshot
+* **Method:** _POST_
+* **Endpoint**:`/volumes/{volume_uuid}/snapshot`
```

Comment:
@obnoxxx this lists `/volumes/{volume_uuid}/snapshot`, in your WIP implementation you used `.../snapshots` (plural). Which would be best? Maybe at one point we would like to list all snapshots of a particular volume (that has been omitted in this design for now). The proposed "Remove all Snapshots of a Volume" might suggest that `.../snapshots` is more suitable.